vp/devel/python_peripherals/DLA.py

```diff
@@ -279,36 +280,12 @@ def construct(flat, shape):
         return [construct(flat[i * sub_size: (i + 1) * sub_size], shape[1:]) for i in range(shape[0])]
 
     output = zeroes(shape)
-    assert get_size(output) == get_size(tensor)
+    assert get_size(output) == get_size(tensor), "Assert failed! Reshape incompatible dimensions"
     flat = flatten(tensor)
 
     return construct(flat, shape)
 
 
-def flatten_tensor(data):
-    """Expect tensor in data CWH format and return 1d array"""
-    in_height = len(data[0][0])
-    in_width = len(data[0])
-    in_channels = len(data)
-    output = []
-    for ch in range(in_channels):
-        for w in range(in_width):
-            for h in range(in_height):
-                output.append(data[ch][w][h])
-    return output
-
-def flat_to_CWH(data, channels, width, height):
-    """Takes in 1d array of length C*W*H and reshapes it to tensor of format CWH"""
-    assert channels * width * height == len(data)
-    output = [[[0 for _ in range(height)] for _ in range(width)] for _ in range(channels)]
-    i = 0
-    for ch in range(channels):
-        for w in range(width):
-            for h in range(height):
-                output[ch][w][h] = data[i]
-                i = i + 1
-    return output
-
 def cast_long_to_signed_byte(value):
```

Comment:
Not part of the review, but I wonder, how did this C-language ABI-specific type names (long) end up in our source tree :sweat_smile: 

Ideally, this would be something like `cast_u64_to_i8`. Besides, current function name leaves it up to interpretation whether it's a signed or an unsigned long. One has to wonder, why is the `signed` name-specifier included for the byte but not for the long.

Reply:
So the logic is that the input value is a Python long type, an integer of (almost) unlimited size. We want to sanitize the longs from Renode and reinterpret them to logical values in the range of -128..127. For naming, i8 would make sense here, but u64 is wrong. I don't know if this answers your question, but it's a necessary evil from working on a byte level in Python with Renode.